src/main/java/io/supertokens/config/CoreConfig.java

```diff
@@ -98,6 +117,43 @@ public String getBasePath() {
         return base_path;
     }
 
+    public enum PASSWORD_HASHING_ALG {
+        ARGON2, BCRYPT
+    }
+
+    public int getArgon2HashingPoolSize() {
```

Comment:
Why do we create the `blockedqueue` in this case? It's fine either way, it just seems unnecessary.

Reply:
Cause it's the simplest way I found to achieve the following:
- We want to run a max of X number of hashes in parallel whilst keeping other hashing requests in a queue.
- A blockedqueue enables this exact use case. When we call `this.boundedQueue.put`, it will block the calling thread if there is no empty space available in the queue (which is of size X).
- It will then unblock the thread as soon as there is space for it in the queue.
- We use the `waiter` object to fill in space in the queue and this object is unique for each thread.

Reply:
I agree. What I meant to ask was why do we normalize the pool size here and create the queue even if it's not used.

Reply:
Ahh. I had first made the queue so that it was null by default and gets initiated only if argon2 is being used, but some fo the tests change the password hashing alg being used without restarting the core in which case I would get a null pointer exception. 

Sure, I could have also initiated the queue in those tests, but initialising it all the time is fine too (it doesn't take up any memory upfront anyway). So for cleaner tests, i did it this way.